lib/inventory_sync/async/inventory_full_sync.rb

```diff
@@ -5,16 +5,16 @@ module Async
     class InventoryFullSync < ::ApplicationJob
       def perform(organization)
         @organization = organization
+        @subscribed_hosts_ids = Set.new(
+          ForemanInventoryUpload::Generators::Queries.for_slice(Host).pluck(:id)
+        )
         @host_statuses = {
           sync: 0,
           disconnect: 0,
         }
-        @all_hosts = Set.new(
-          ForemanInventoryUpload::Generators::Queries.for_slice(Host.unscoped).pluck(:id)
-        )
 
         InventorySync::InventoryStatus.transaction do
-          InventorySync::InventoryStatus.delete_all
+          InventorySync::InventoryStatus.where(host_id: @subscribed_hosts_ids).delete_all
```

Comment:
We shouldn't delete other orgs statuses as they won't be created again in this job.